File: FoosballGames/Bootstrap.cs
using FoosballGames.Contracts;
using FoosballGames.Infrastructure.Messaging;
using Microsoft.Extensions.DependencyInjection;

namespace FoosballGames
{
    public static class Bootstrap
    {
        public static IServiceCollection RegisterFoosballGamesComponents(this IServiceCollection serviceCollection)
        {
            return serviceCollection
                .AddScoped<IQueryHandler<GetFoosballGames, FoosballGamesResponse>, FoosballGamesQueryHandler>()
                .AddScoped<IQueryHandler<GetFoosballGame, Contracts.FoosballGame>, FoosballGamesQueryHandler>()
                .AddScoped<ICommandHandler<AddPointForTeam>, FoosballGameCommandHandler>()
                .AddScoped<ICommandHandler<CreateFoosballGame>, FoosballGameCommandHandler>();
        }
    }
}

File: FoosballGames/Bootstrap.fs
module FoosballGames.Bootstrap

open FoosballGames.Contracts
open FoosballGames.Infrastructure.Messaging
open Microsoft.Extensions.DependencyInjection

type IServiceCollection with
    member this.RegisterFoosballGamesComponents() =
        this
            .AddScoped<IQueryHandler<GetFoosballGames, FoosballGamesResponse>, FoosballGamesQueryHandler>()
            .AddScoped<IQueryHandler<GetFoosballGame, Contracts.FoosballGame>, FoosballGamesQueryHandler>()
            .AddScoped<ICommandHandler<AddPointForTeam>, FoosballGameCommandHandler>()
            .AddScoped<ICommandHandler<CreateFoosballGame>, FoosballGameCommandHandler>();
    